core-bundle/src/Resources/contao/classes/DataContainer.php

```diff
@@ -953,7 +953,7 @@ protected function generateButtons($arrRow, $strTable, $arrRootIds=array(), $bln
 					if (($params['act'] ?? null) == 'toggle' && isset($params['field']))
 					{
 						// Hide the toggle icon if the user does not have access to the field
-						if (!System::getContainer()->get('security.helper')->isGranted(ContaoCorePermissions::USER_CAN_EDIT_FIELD_OF_TABLE, $this->strTable . '::' . $params['field']))
+						if (($GLOBALS['TL_DCA'][$strTable]['fields'][$params['field']]['toggle'] ?? false) !== true || !System::getContainer()->get('security.helper')->isGranted(ContaoCorePermissions::USER_CAN_EDIT_FIELD_OF_TABLE, $strTable . '::' . $params['field']))
```

Comment:
Isn't this a BC break? Because the condition will also be true if the field has no `toggle` property, which is the case in Contao <4.13.